_appmap/test/test_command.py

```diff
@@ -42,40 +43,44 @@ class TestAgentStatus:
     @pytest.mark.parametrize("do_discovery", [True, False])
     def test_test_discovery_control(self, cmd_setup, do_discovery, mocker):
         mocker.patch("appmap.command.appmap_agent_status.discover_pytest_tests")
-        rc = appmap_agent_status._run(discover_tests=do_discovery)
+        rc = appmap_agent_status._run(  # pylint: disable=protected-access
+            discover_tests=do_discovery
+        )
         assert rc == 0
         call_count = 1 if do_discovery else 0
         assert appmap_agent_status.discover_pytest_tests.call_count == call_count
 
     @pytest.mark.parametrize("cmd_setup", ["pytest"], indirect=True)
     def test_agent_status(self, cmd_setup, capsys):
-        rc = appmap_agent_status._run(discover_tests=True)
+        rc = appmap_agent_status._run(discover_tests=True)  # pylint: disable=protected-access
+
         assert rc == 0
         output = capsys.readouterr()
         status = json.loads(output.out)
         # XXX This will detect pytest as the test framework, because
         # appmap-python uses it. We need a better mechanism to handle
         # testing more broadly.
-        assert status == DictIncluding(
+        props = status["properties"]
+        config = props["config"]
+        assert config == DictIncluding({"app": "Simple", "present": True, "valid": True})
+        project = props["project"]
+        assert project == DictIncluding(
             {
-                "properties": {
-                    "config": {"app": "Simple", "present": True, "valid": True},
-                    "project": {
-                        "agentVersion": "0.0.0",
-                        "language": "python",
-                        "remoteRecordingCapable": True,
-                        "integrationTests": True,
-                    },
-                },
-                "test_commands": [
-                    {"args": [], "framework": "pytest", "command": "pytest"}
-                ],
+                "language": "python",
+                "remoteRecordingCapable": True,
+                "integrationTests": True,
             }
         )
+        assert "agentVersion" in project
+        test_commands = status["test_commands"]
+        assert test_commands[0] == DictIncluding(
+            {"args": [], "framework": "pytest", "command": "pytest"}
+        )
```

Comment:
Is there a reason to break this out like that?

Reply:
It was originally checking `status` against
```
            {
                "properties": {
                    "config": {"app": "Simple", "present": True, "valid": True},
                    "project": {
                        "agentVersion": "0.0.0",
                        "language": "python",
                        "remoteRecordingCapable": True,
                        "integrationTests": True,
                    },
                },
                "test_commands": [
                    {"args": [], "framework": "pytest", "command": "pytest"}
                ],
                "language": "python",
                "remoteRecordingCapable": True,
                "integrationTests": True,
            }
```
This was failing, because status has a valid value for `agentVersion`, which doesn't match `"0.0.0"`, of course.

When I tried pulling `agentVersion` out of `project`, the comparison still failed, because `status` had it, but the expected value didn't.

I assumed that the `dict` `>=` comparison just wasn't doing the right thing with nested `dict`s, so I unnested them.

Reply:
Ah, ok, makes sense. That's fine, thanks. (Though I wonder if `"project": DictIncluding(...)` would work instead?)